hathor/p2p/netfilter/factory.py

```diff
@@ -31,7 +31,7 @@ def __init__(self, connections: 'ConnectionsManager', wrappedFactory: 'IProtocol
         super().__init__(wrappedFactory)
         self.connections = connections
 
-    def buildProtocol(self, addr: IAddress) -> Protocol:
+    def buildProtocol(self, addr: IAddress) -> Optional[Protocol]:
```

Comment:
Optional?

Reply:
I think this came up because of a superclass, but I'll revise it.

Reply:
I just revised it, it's not because of a superclass, we literally return `None` in the body of the function, so the de-facto type is `Optional`. Is it a mistake to return `None`?

Reply:
Oh! You're right. We do it to refuse a new connection. Thanks!